### DISCUSSION TITLE
Updating to v6 breaks Material-UI's Autocomplete, component doesn't update field value, state always return integer 0

### DISCUSSION THREAD

**Discussion by markcorrea** [react-hook-form/react-hook-form#2051] [Q&A] [answered]
**Describe the bug**
After I updated to `react-hook-form v6`, my `Material-UI`'s `Autocomplete` components broke in all pages. They are unable to select any value. The console shows me the error on the screenshot below and, after trying to change a field, if I check the values in the `getValues`, they are always integer `0`. No matter which option I select, they just change the value to `0`. 

I've tried to add the `onChange` function, but it seems to make no difference at all. I've also tried to add the Autocomplete's  `renderOption`, but also doesn't make difference. It points that there's an error with the `getOptionLabel` attribute, but it will always, on every attempt, return `0` and add it to this field's value.

**To Reproduce**
https://codesandbox.io/s/flamboyant-brown-dcmxm?file=/src/App.js
I've created this small sandbox with the error:
- Select any choice inside the Autocomplete;
- click the button `CONSOLE VALUES` (this will log in the console the options we have on the form's `getValues()` function);
- Go to the console;

**Expected behavior**
- The `Autocomplete` will change to the desired option and the object on the console will display an object with the name of the USA selected state. example: {state: "alabama"}

**Current behavior**
- No matter which option I choose in the `Autocomplete`, the state will always change to integer `0`. 

**Screenshots**
<img width="210" alt="Screen Shot 2020-07-03 at 16 48 08" src="https://user-images.githubusercontent.com/12589403/86494957-aebd6280-bd4d-11ea-9a11-f71344113e86.png">

Is anyone else having issues like this in the sandbox? Have you come to any solutions?

**Comment by bluebill1049** [comment, marked as answer]
hey @markcorrea there is breaking change with `Controller`.  https://github.com/react-hook-form/react-hook-form/blob/master/CHANGELOG.md

we have removed the following props: 

- onChange
- onChangeName
- onBlur
- onBlurName
- valueName

and in favor of `render` prop

```diff
-<Controller
-  as={CustomInput}
-  valueName="textValue"
-  onChangeName="onTextChange"
-  control={control}
-  name="test"
-/>
+<Controller
+  render={({ onChange, onBlur, value }) => (
+     <CustomInput onTextChange={onChange} onBlur={onBlur} textValue={value} />
+  )}
+  control={control}
+  name="test"
+/>
```

**Comment by markcorrea** [comment]
Noted! I haven't noticed this breaking change. Will update the component now. 
Thanks a lot for the reply!

**Reply by bluebill1049** [reply]
No worries, we have a CHANGELOG https://github.com/react-hook-form/react-hook-form/blob/master/CHANGELOG.md you can keep an eye on that.
